Fixes the typing rule of CAUGHTadm (#244)
WebAssembly/exception-handling: document/core/appendix/properties.rst

* The :ref:`values <syntax-val>` :math:`\val^\ast` must be of type :math:`[t_0^\ast]`.

* The label :math:`C.\CLABELS[0]` must be defined in the context.

* Let :math:`(\LCATCH^?~[t^\ast])` be the :ref:`label type <syntax-labeltype>` :math:`C.\CLABELS[0]`.

> * Let :math:`C'` be the same :ref:`context <context>` as :math:`C`, but with the first label popped from the |CLABELS| vector.
> 
> * Let :math:`C''` be the same :ref:`context <context>` as :math:`C'`, but with the label type :math:`(\LCATCH~[t^\ast])` prepended to the |CLABELS| vector.

Reviewer: I think you can merge these:

```suggestion
* Let :math:`C''` be the same :ref:`context <context>` as :math:`C`, but with the label type :math:`(\LCATCH~[t^\ast])` replacing the first element of the |CLABELS| vector.
```

Author: For some reason I couldn't apply the suggestions but I added these changes to the latest commit I just pushed.

* Under context :math:`C''`,
  the instruction sequence :math:`\instr^\ast` must be :ref:`valid <valid-instr-seq>` with type :math:`[] \to [t^\ast]`.

* Then the compound instruction is valid with type :math:`[] \to [t^\ast]`.

.. math::
   \frac{
     S \vdashexternval \EVTAG~\tagaddr : \ETTAG~[t_0^\ast]\to[]
     \qquad
     (val : t_0)^\ast
     \qquad
     S; C',\CLABELS\,(\LCATCH~[t^\ast]) \vdashinstrseq \instr^\ast : [] \to [t^\ast]
   }{

>      S; C',\CLABELS\,(\LCATCH^?~[t^\ast]) \vdashadmininstr \CAUGHTadm\{\tagaddr~\val^\ast\}~\instr^\ast~\END : [] \to [t^\ast]

Reviewer: I don't think we can or want to allow this to be a catch label beforehand already.
```suggestion
     S; C',\CLABELS\,[t^\ast] \vdashadmininstr \CAUGHTadm\{\tagaddr~\val^\ast\}~\instr^\ast~\END : [] \to [t^\ast]
```

Author: Same as above
